ext/node/polyfills/internal_binding/stream_wrap.ts

```diff
@@ -311,21 +311,36 @@ export class LibuvStreamWrap extends HandleWrap {
 
   /** Internal method for reading from the attached stream. */
   async #read() {
-    const isOwnedBuf = bufLocked;
-    let buf = bufLocked ? new Uint8Array(SUGGESTED_SIZE) : BUF;
-    bufLocked = true;
+    // Lock safety: We must hold this lock until we are certain that buf is no longer used
+    // This setup code is a little verbose, but we need to be careful about buffer management
+    let buf, locked;
```

Comment:
**nitpick:** I'd prep locked to false. I do not know if that helps V8 with optimisation but it might. Also makes TS understand it even in a JS file.